airflow/providers/hashicorp/common/vault_client.py

```diff
@@ -0,0 +1,389 @@
+        elif self.auth_type == 'azure':
+            _client.auth.azure.configure(
+                tenant_id=self.azure_tenant_id,
+                resource=self.azure_resource,
+                client_id=self.key_id,
+                client_secret=self.secret_id,
+                mount_point=self.mount_point
+            )
+        elif self.auth_type == "gcp":
+            # noinspection PyProtectedMember
+            from airflow.providers.google.cloud.utils.credentials_provider import (
+                get_credentials_and_project_id,
+                _get_scopes
+            )
+            scopes = _get_scopes(self.gcp_scopes)
+            credentials, _ = get_credentials_and_project_id(key_path=self.gcp_key_path,
+                                                            keyfile_dict=self.gcp_keyfile_dict,
+                                                            scopes=scopes)
+            _client.auth.gcp.configure(credentials=credentials, mount_point=self.mount_point)
+        elif self.auth_type == "github":
+            _client.auth.github.login(token=self.token, mount_point=self.mount_point)
+        elif self.auth_type == "kubernetes":
+            if not self.kubernetes_jwt_path:
+                raise VaultError("The kubernetes_jwt_path should be set here. This should not happen.")
+            with open(self.kubernetes_jwt_path) as f:
+                jwt = f.read()
+                _client.auth_kubernetes(role=self.kubernetes_role, jwt=jwt,
+                                        mount_point=self.mount_point)
+        elif self.auth_type == "ldap":
+            _client.auth.ldap.login(
+                username=self.username, password=self.password, mount_point=self.mount_point)
+        elif self.auth_type == "radius":
+            _client.auth.radius.configure(host=self.radius_host,
+                                          secret=self.radius_secret,
+                                          port=self.radius_port,
+                                          mount_point=self.mount_point)
+        elif self.auth_type == "token":
+            _client.token = self.token
+        elif self.auth_type == "userpass":
+            _client.auth_userpass(username=self.username, password=self.password,
+                                  mount_point=self.mount_point)
+        else:
+            raise VaultError(f"Authentication type '{self.auth_type}' not supported")
+
+        if _client.is_authenticated():
+            return _client
+        else:
+            raise VaultError("Vault Authentication Error!")
+
+    def get_secret(self, secret_path: str, secret_version: Optional[int] = None) -> Optional[dict]:
+        """
+        Get secret value from the engine.
+
+        :param secret_path: path of the secret
+        :type secret_path: str
+        :param secret_version: optional version of key to read - can only be used in case of version 2 of KV
```

Comment:
```suggestion
        :param secret_version: Specifies the version of Secret to return. If not set the latest version is returned. (Can only be used in case of version 2 of KV)
```